// secure-env/dist/es5/lib/utils/log.js
"use strict";

var colorCodes = {
  BLACK: '\x1b[30m',
  RED: '\x1b[31m',
  GREEN: '\x1b[32m',
  YELLOW: '\x1b[33m',
  BLUE: '\x1b[34m',
  MAGENTA: '\x1b[35m',
  CYAN: '\x1b[36m',
  RESET: '\x1b[0m'
};

function getColorCode(colorName) {
  var colorCode = colorCodes[colorName.toUpperCase()];
  return colorCode || colorCodes.RESET;
}

module.exports = function (data, type, color) {
  if (data) {
    var logData = "".concat(colorCodes.CYAN, "Secure-env : ").concat(colorCodes.RESET);

    switch (type.toUpperCase()) {
      case 'ERROR':
        logData += "".concat(colorCodes.RED, " ERROR OCCURED ").concat(colorCodes.RESET);
        break;

      case 'INFO':
        logData += "".concat(colorCodes.GREEN, " INFO ").concat(colorCodes.RESET);
        break;

      case 'WARN':
        logData += "".concat(colorCodes.YELLOW, " WARNING ").concat(colorCodes.RESET);
        break;
    }

    logData += color ? "".concat(getColorCode(color)).concat(data).concat(colorCodes.RESET) : data;
    console.log(logData);
  }
};
